django/utils/dateformat.py

```diff
@@ -265,16 +265,13 @@ def o(self):
 
     def r(self):
         "RFC 5322 formatted date; e.g. 'Thu, 21 Dec 2000 16:01:07 +0200'"
-        if type(self.data) is date:
-            raise TypeError(
-                "The format for date objects may not contain time-related "
-                "format specifiers (found 'r')."
-            )
-        if is_naive(self.data):
-            dt = make_aware(self.data, timezone=self.timezone)
-        else:
-            dt = self.data
-        return format_datetime_rfc5322(dt)
+        value = self.data
+        if not isinstance(value, datetime):
+            # Assume midnight UTC if instance of datetime.date provided.
+            value = datetime.combine(value, time.min).replace(tzinfo=timezone.utc)
```

Comment:
Why not the `get_default_timezone()` instead? :thinking:

Reply:
Think I just tend to default to UTC. But, you're right, default timezone would make more sense here.

Also checked how PHP behaves given that this implementation is largely based on theirs and the following:
```php
<?php
date_default_timezone_set('Australia/Sydney');
echo(date("r", 0));
```
Yields the following:
```
Thu, 01 Jan 1970 10:00:00 +1000
```
Will switch to use `get_default_timezone()` 👍🏻

Reply:
Updated.